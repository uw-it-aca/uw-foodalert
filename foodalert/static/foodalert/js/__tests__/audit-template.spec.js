"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
import Vuelidate from 'vuelidate'
Vue.use(Vuelidate)
Vue.use(BootstrapVue);

import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
Vue.component('font-awesome-icon', FontAwesomeIcon);

import { library } from '@fortawesome/fontawesome-svg-core'
import {
    faUtensils,
    faChevronLeft,
    faCalendar,
    faClock,
    faMapMarkerAlt,
    faClipboardCheck,
    faBell,
} from '@fortawesome/free-solid-svg-icons'

library.add(faUtensils)
library.add(faChevronLeft)
library.add(faCalendar)
library.add(faClock)
library.add(faBell)
library.add(faMapMarkerAlt)
library.add(faClipboardCheck)

import {mount} from '@vue/test-utils';
import AuditTemplate from '../components/audit-template.vue';

describe('Audit Template', function () {
    describe('renders correctly', function () {
        test('textarea', function () {
            var wrapper = mount(AuditTemplate, {
                propsData: {
                    items: [{ dateSent: '10/22/18', timeSent: '4:44PM', endTime: '5:32PM', updateTime: '5:10PM', host: 'netid@uw.edu', permitNumber: '12345678', foodAndEvent: 'A Food and Event', quantity: 'About 8 meals', location: 'HUB 101', contains: 'Dairy, Meat' }]
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        });
    });
})
