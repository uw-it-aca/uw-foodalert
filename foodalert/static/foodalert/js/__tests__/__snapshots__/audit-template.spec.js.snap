// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit Controller renders correctly textarea 1`] = `
<div>
  <div>
    <div
      class="btn-group b-dropdown dropdown"
      id="__BVID__3"
    >
      <!---->
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-secondary dropdown-toggle"
        id="__BVID__3__BV_toggle_"
        type="button"
      >
        Year
      </button>
      <div
        aria-labelledby="__BVID__3__BV_toggle_"
        class="dropdown-menu"
        role="menu"
      >
        <a
          class="dropdown-item"
          href="#"
          role="menuitem"
          target="_self"
        >
          
           All
           
        </a>
        <a
          class="dropdown-item"
          href="#"
          role="menuitem"
          target="_self"
        >
          
           2018
           
        </a>
      </div>
    </div>
     
    <div
      class="btn-group b-dropdown dropdown"
      id="__BVID__4"
    >
      <!---->
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-secondary dropdown-toggle"
        id="__BVID__4__BV_toggle_"
        type="button"
      >
        Month
      </button>
      <div
        aria-labelledby="__BVID__4__BV_toggle_"
        class="dropdown-menu"
        role="menu"
      >
        <a
          class="dropdown-item"
          href="#"
          role="menuitem"
          target="_self"
        >
          
           All
           
        </a>
        <a
          class="dropdown-item"
          href="#"
          role="menuitem"
          target="_self"
        >
          
           Dec
           
        </a>
      </div>
    </div>
     
    <button
      class="btn btn-secondary"
      style="float: right;"
      type="button"
    >
      Export
    </button>
  </div>
   
  <p />
   
  <table
    aria-busy="false"
    aria-colcount="12"
    aria-rowcount="2"
    class="table b-table table-hover"
    id="__BVID__5"
  >
    <!---->
    <!---->
    <thead
      class=""
    >
      <tr>
        <th
          aria-colindex="1"
          class=""
        >
          Location
        </th>
        <th
          aria-colindex="2"
          class=""
        >
          Event
        </th>
        <th
          aria-colindex="3"
          class=""
        >
          Time Created
        </th>
        <th
          aria-colindex="4"
          class=""
        >
          Time Ended
        </th>
        <th
          aria-colindex="5"
          class=""
        >
          Food Served
        </th>
        <th
          aria-colindex="6"
          class=""
        >
          Food Amount
        </th>
        <th
          aria-colindex="7"
          class=""
        >
          Food Allergens
        </th>
        <th
          aria-colindex="8"
          class=""
        >
          Bring Containers
        </th>
        <th
          aria-colindex="9"
          class=""
        >
          Host Net ID
        </th>
        <th
          aria-colindex="10"
          class=""
        >
          Host User Agent
        </th>
        <th
          aria-colindex="11"
          class=""
        >
          Permit Number
        </th>
        <th
          aria-colindex="12"
          class=""
        >
          Host ID
        </th>
      </tr>
    </thead>
    <!---->
    <tbody
      class=""
    >
      <!---->
      <tr
        class=""
      >
        <td
          aria-colindex="1"
          class=""
        >
          HUB 130
        </td>
        <td
          aria-colindex="2"
          class=""
        >
          Placeholder event
        </td>
        <td
          aria-colindex="3"
          class=""
        >
          Tue Dec 18 2018 10:20:00 AM
        </td>
        <td
          aria-colindex="4"
          class=""
        >
          Tue Dec 18 2018 11:11:00 AM
        </td>
        <td
          aria-colindex="5"
          class=""
        >
          a
        </td>
        <td
          aria-colindex="6"
          class=""
        >
          a
        </td>
        <td
          aria-colindex="7"
          class=""
        />
        <td
          aria-colindex="8"
          class=""
        >
          false
        </td>
        <td
          aria-colindex="9"
          class=""
        >
          
        </td>
        <td
          aria-colindex="10"
          class=""
        >
          Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:64.0) Gecko/20100101 Firefox/64.0
        </td>
        <td
          aria-colindex="11"
          class=""
        >
          2
        </td>
        <td
          aria-colindex="12"
          class=""
        >
          1
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          aria-colindex="1"
          class=""
        >
          HUB 130
        </td>
        <td
          aria-colindex="2"
          class=""
        >
          Placeholder event
        </td>
        <td
          aria-colindex="3"
          class=""
        >
          Tue Dec 18 2018 11:08:04 AM
        </td>
        <td
          aria-colindex="4"
          class=""
        >
          Tue Dec 18 2018 1:00:00 PM
        </td>
        <td
          aria-colindex="5"
          class=""
        >
          Leftover food from HUB meeting
        </td>
        <td
          aria-colindex="6"
          class=""
        >
          About 8 full meals
        </td>
        <td
          aria-colindex="7"
          class=""
        />
        <td
          aria-colindex="8"
          class=""
        >
          false
        </td>
        <td
          aria-colindex="9"
          class=""
        >
          
        </td>
        <td
          aria-colindex="10"
          class=""
        >
          Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:64.0) Gecko/20100101 Firefox/64.0
        </td>
        <td
          aria-colindex="11"
          class=""
        >
          3
        </td>
        <td
          aria-colindex="12"
          class=""
        >
          1
        </td>
      </tr>
      <!---->
      <!---->
    </tbody>
  </table>
</div>
`;
